content/en/docs/Core concepts/baker_components.md

```diff
@@ -0,0 +1,135 @@
+---
+title: "baker.Components"
+date: 2020-10-29
+weight: 3
+description: >
+  baker.Components is the main object used to create a Baker topology
+---
+
+`baker.Components` is a struct that is used to configure the Baker topology, before running it.
+
+It contains the available components and some functions used by Baker to deal with records.
+
+```go
+type Components struct {
+	Inputs         []InputDesc
+    Filters        []FilterDesc
+	Outputs        []OutputDesc
+    Uploads        []UploadDesc
+    
+	Metrics        []MetricsDesc
+	User           []UserDesc
+    
+    ShardingFuncs  map[FieldIndex]ShardingFunc
+	Validate       ValidationFunc
+	CreateRecord   func() Record
+	FieldByName    func(string) (FieldIndex, bool)
+	FieldName      func(FieldIndex) string
+}
+```
+
+## Inputs, Filters, Outputs and Uploads
+
+These fields contain the list of components that are available to the topology.
```

Comment:
I like the idea of a more structured page where we can add more details, like in this case, instead of a list with short comments. Please check now after the new commits